healthcheck/dial.go

```diff
@@ -51,112 +50,30 @@ func (c *conn) Close() error {
 	return nil
 }
 
-func sockaddrToString(sa syscall.Sockaddr) string {
-	switch sa := sa.(type) {
-	case *syscall.SockaddrInet4:
-		return net.JoinHostPort(net.IP(sa.Addr[:]).String(), strconv.Itoa(sa.Port))
-	case *syscall.SockaddrInet6:
-		return net.JoinHostPort(net.IP(sa.Addr[:]).String(), strconv.Itoa(sa.Port))
-	default:
-		return fmt.Sprintf("(unknown - %T)", sa)
+func (conn *DialConn) control(network, address string, c syscall.RawConn) error {
+	ctl := func(fd uintptr) {
+		if conn.mark != 0 {
+			if err := setSocketMark(int(fd), conn.mark); err != nil {
+				return
```

Comment:
the error is hidden. can we have a variable to store it?

var fdErr error
ctl := func(fd uintptr) {.. fdErr = xxx;}

if err := c.Control(ctl); err != nil {
}

return fdErr

Reply:
good suggestion. I will fix.